### ISSUE TITLE
Follower should be able to vote more than once in a term

### ISSUE THREAD

**Issue by schuster** [ktoso/akka-raft#29]
Because Raft RPCs are idempotent, Followers should be able to vote for the same candidate more than once in a given term. This allows the candidate to receive a response in case the first response was dropped.

**Comment by colin-scott** [comment]
For what it's worth, I have a (non-pull-request-worthy) fix for this here:

https://github.com/NetSys/sts2-applications/commit/4e733c46db4568b18d9240b7a4654f46d70787ae
https://github.com/NetSys/sts2-applications/commit/7fecff5925e76dd0e27c672089e4ca580d11b8db7fecff5925e76dd0e27c672089e4ca580d11b8db

**Comment by dmitraver** [comment]
@colin-scott Are you planning to make a pull request with all your changes?

**Comment by colin-scott** [comment]
Yeah, @ktoso just suggested that I should. But some of them do depend on your #55, for what it's worth.

**Comment by ktoso** [comment]
I'll look into #55 and others tonight so we can start rolling fixes in then :)

**Comment by dmitraver** [comment]
@ktoso Good news :) I'm looking forward to that
